src/view/context/view.cpp

```diff
@@ -24,32 +65,48 @@ void Context::View::updateContextIcon()
 
 void Context::View::resetCurrentlyPlaying() const
 {
-	if (content != nullptr)
+	if (albumShouldBeExpandable && horizContent != nullptr)
+	{
+		horizContent->reset();
+	}
+	else if(!albumShouldBeExpandable && content != nullptr)
 	{
 		content->reset();
 	}
 }
 
 auto Context::View::getCurrentlyPlaying() const -> const lib::spt::track &
 {
-	if (content == nullptr)
+	if (albumShouldBeExpandable && horizContent != nullptr)
+	{
+		return horizContent->getCurrentlyPlaying();
+	}
+	else if(!albumShouldBeExpandable && content != nullptr)
 	{
-		throw std::runtime_error("No content");
+		return content->getCurrentlyPlaying();
 	}
-	return content->getCurrentlyPlaying();
+	throw std::runtime_error("No content");
 }
 
 void Context::View::setCurrentlyPlaying(const lib::spt::track &track) const
 {
-	if (content != nullptr)
+	if (albumShouldBeExpandable && horizContent != nullptr)
```

Comment:
Having a common interface for both of these would make this a lot prettier, but it's not a requirement. Did you try using pure virtual methods?

Reply:
Yea - I tried to do this using [this stackoverflow post](https://stackoverflow.com/questions/9756893/how-to-implement-interfaces-in-c) but got a ton of obtuse linker errors. The code would compile - but then the linker would complain. Something about "vtable not found".

I'll give it another shot!

Reply:
Don't know how helpful it is, but there's a similar thing with paths, where I have a generic interface for the methods:
https://github.com/kraxarn/spotify-qt/blob/894e220356183344043b74d43ab696a04e21ed25/lib/include/lib/paths/paths.hpp#L13
And then an implementation for Qt:
https://github.com/kraxarn/spotify-qt/blob/894e220356183344043b74d43ab696a04e21ed25/src/lib/qtpaths.hpp#L8